pkg/providers/amifamily/resolver.go

```diff
@@ -117,17 +119,20 @@ func New(amiProvider *Provider) *Resolver {
 // Multiple ResolvedTemplates are returned based on the instanceTypes passed in to support special AMIs for certain instance types like GPUs.
 func (r Resolver) Resolve(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, machine *v1alpha5.Machine, instanceTypes []*cloudprovider.InstanceType, options *Options) ([]*LaunchTemplate, error) {
 	amiFamily := GetAMIFamily(nodeTemplate.Spec.AMIFamily, options)
-	amis, err := r.amiProvider.Get(ctx, nodeTemplate, options)
+	mappedAMIs, err := r.getMappedAMIs(ctx, nodeTemplate, instanceTypes, options)
```

Comment:
Really nice code simplification 🎉  I'd still recommend that we do this change separately from this PR so that all the code changes make sense in context. I'm not sure this change was needed in order to get the bug fix in.

Reply:
I agree it doesn't really make sense in this PR. I included it because I was failing tests due to function complexity and this change got it through. I can try getting the structured version of the kubeVersion directly and see if that makes a difference.

Reply:
Ah, that makes more sense 👍 If that's blocking, this is fine to me

Reply:
Might be helpful in the future to self-comment this on the PR so the change is well-understood